src/pdm/models/session.py

```diff
@@ -134,15 +134,18 @@ def __init__(self, *, sources: list[RepositoryConfig], cache_dir: Path, **kwargs
         self._trusted_host_ports: set[tuple[str, int | None]] = set()
         transport: httpx.BaseTransport | None = None
         for s in sources:
+            url = None
+            if s.url is not None:
```

Comment:
this check is unnecessary. we just assert source.URL is non-null

Reply:
Is there a case where a `pypi` source doesn't have a URL? If so I'm not sure if it would make sense to add it to the `_trusted_host_ports` since there's no URL to add

Reply:
yes. but not in this specific function. it always have a URL for a source